runtime/common/src/test/kotlin/kotlinx/serialization/features/LongAsStringTest.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.serialization.features
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.internal.LongAsStringSerializer
+import kotlinx.serialization.json.JsonTestBase
+import kotlin.test.*
+
+class LongAsStringTest : JsonTestBase() {
+    @Serializable
+    data class HasLong(@Serializable(LongAsStringSerializer::class) val l: Long)
+
+    @Test
+    @Ignore // this needs 1.3.20-eap-1 due to #252. Tested locally on snapshot.
+    fun canSerializeAsStringAndParseBack() = parametrizedTest { useStreaming ->
+        val original = HasLong(Long.MAX_VALUE - 1)
+        val str = strict.stringify(HasLong.serializer(), original, useStreaming)
+        assertEquals("""{"l":"9223372036854775806"}""", str)
+        val restored = strict.parse(HasLong.serializer(), str, useStreaming)
+        assertEquals(original, restored)
+    }
+
+    @Test
+    fun canNotDeserializeInvalidString() = parametrizedTest { useStreaming ->
+        val str = """{"l": "this is definitely not a long"}"""
+        assertFailsWith<NumberFormatException> { strict.parse(HasLong.serializer(), str, useStreaming) }
```

Comment:
Minor suggestion: add a test for number literal, which doesn't fit into a Long, e.g. `1000000000000000000000` Just to make sure the exception makes sense (and that it is actually raised).